src/expandableCalendar/WeekCalendar/index.js

```diff
@@ -102,79 +102,46 @@ class WeekCalendar extends Component {
   });
 
   onDayPress = value => {
-    _.invoke(this.props.context, 'setDate', value.dateString, UPDATE_SOURCES.DAY_PRESS);
+    this.presenter.onDayPressed(this.props.context, value);
   };
 
   onScroll = ({
     nativeEvent: {
       contentOffset: {x}
     }
   }) => {
-    if (!this.firstAndroidRTLScrollIgnored) {
-      this.firstAndroidRTLScrollIgnored = true;
-      return;
-    }
-
-    // Fix reversed offset on Android+RTL
-    if (applyAndroidRtlFix) {
-      const numOfPages = this.state.items.length - 1;
-      const overallWidth = numOfPages * this.containerWidth;
-      x = overallWidth - x;
-    }
-
-    const newPage = Math.round(x / this.containerWidth);
+    const {onScroll} = this.presenter;
+    const {context} = this.props;
+    const {items} = this.state;
+    const {containerWidth: width, page} = this;
 
-    if (this.page !== newPage) {
-      const {items} = this.state;
+    const updateState = (newData, newPage) => {
       this.page = newPage;
+      this.setState({items: [...newData]});
+    };
 
-      _.invoke(this.props.context, 'setDate', items[this.page], UPDATE_SOURCES.WEEK_SCROLL);
-
-      if (this.page === items.length - 1) {
-        for (let i = 0; i <= NUMBER_OF_PAGES; i++) {
-          items[i] = items[i + NUMBER_OF_PAGES];
-        }
-        this.setState({items: [...items]});
-      } else if (this.page === 0) {
-        for (let i = items.length - 1; i >= NUMBER_OF_PAGES; i--) {
-          items[i] = items[i - NUMBER_OF_PAGES];
-        }
-        this.setState({items: [...items]});
-      }
-    }
+    onScroll({context, updateState, x, page, items, width});
   };
 
   onMomentumScrollEnd = () => {
     const {items} = this.state;
-    const isFirstPage = this.page === 0;
-    const isLastPage = this.page === items.length - 1;
-
-    if (isFirstPage || isLastPage) {
-      this.list.current.scrollToIndex({animated: false, index: NUMBER_OF_PAGES});
-      this.page = NUMBER_OF_PAGES;
-      const newWeekArray = this.getDatesArray();
-
-      if (isLastPage) {
-        for (let i = NUMBER_OF_PAGES + 1; i < items.length; i++) {
-          items[i] = newWeekArray[i];
-        }
-      } else {
-        for (let i = 0; i < NUMBER_OF_PAGES; i++) {
-          items[i] = newWeekArray[i];
-        }
-      }
+    const {onMomentumScrollEnd} = this.presenter;
 
+    const updateState = updatedItems => {
```

Comment:
I don't understand why is 'updateState' const is here, passed to the presenter's 'onMomentumScrollEnd'? Why it's not done by the presenter?